### PR TITLE
feat(lpa-submitted-e2e-tests): feature file and step files for as-10

### PR DESCRIPTION
**Pull request by swdpcomputing**
## Ticket Number

AS-1594

## Description of change
added feature file and step files

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+Given('The question "Upload the Planning Officers report" has been completed', () => {
+  cy.goToPage(pageUrl);
+  cy.clickChooseFile(validFileName);
+  cy.clickUploadFile().click();
+  cy.validateFileUpload(validFileName);
+  cy.clickSaveAndContinue();
+});
+
+When('LPA Planning Officer deletes the file', () => {
+  cy.deleteUploadedFile(validFileName);
```
Comment (shilpabasu22) [inline lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:47]: Do we need an API request to check if the file exists before we delete it?

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+Given('a file has been uploaded', () => {
+  cy.goToPage(pageUrl);
+  cy.clickChooseFile(validFileName);
+  cy.clickUploadFile().click();
+  cy.validateFileUpload(validFileName);
+});
+
+Given('a file has been uploaded and confirmed', () => {
+  cy.goToUploadPlanningOfficersReportPage();
```
Comment (shilpabasu22) [inline lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:31]: This is an example of inconsistency that can be caused by deleting the initially written goTo methods for each page. You have deleted this but are still referring it here. Please add it as a separate refactor task and once prioritised, we can edit all the goTo methods at the same time.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+When('files of {string} have been selected', (fileName) => {
+  cy.clickChooseFile(fileName);
+  cy.clickUploadFile().click();
+  cy.validateFileUpload(fileName);
+  cy.clickSaveAndContinue();
+});
+
+When('the Planning Officers report is requested', () => {
+  cy.verifyCompletedStatus(completedStatusId);
```
Comment (shilpabasu22) [inline lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:91]: This step should be about navigating to the Planning Officer reports page and not about verifying the task status

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+|upload-file-valid.jpg|
+|upload-file-valid.jpeg|
+|upload-file-valid.doc|
+|upload-file-valid.docx|
+
+Scenario Outline: AC3 LPA Planning Officer successfully uploads file via Drag and Drop
+Given Upload the Planning Officers report is requested
+When a '<valid_file>' is uploaded via drag and drop
+Then progress is made to the Task List
+And Upload the Planning Officers report is requested
```
Comment (pippalou) [inline lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:31]: Should this be:  And upload the Planning Officer's report subsection is shown as completed @swdpcomputing @shilpabasu22

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+When files of '<invalid_format>' have been selected
+Then progress is halted with error message 'The file must be DOC, DOCX, PDF, TIF, JPG or PNG'
+Examples:
+|invalid_format|
+|upload-file-invalid-wrong-type.csv|
+
+Scenario: AC8 LPA Planning Officer selects to return to previous page
+Given Upload the Planning Officers report is requested
+When Back is then requested
+Then the Task list is presented
```
Comment (pippalou) [inline lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:78]: Do we need to add this: And any document upload will not be saved

**Review by pippalou — CHANGES_REQUESTED** [review]
Ive added a couple of comments 
Pippa

**Review by shilpabasu22 — CHANGES_REQUESTED** [review]

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature [hunk 6]
```diff
@@ -0,0 +1,97 @@
+Feature: uploadReportToReachDecision
+  As a Beta LPA Planning Officer
+  I want to provide the Planning Inspectorate with a copy of the Planning Officers Report
+  So that this can form part of the Inspectors decision
+
+Scenario: AC1 Access Upload the Planning Officers report question
+Given LPA Planning Officer is within the LPA Questionnaire task list
+When LPA Planning Officer selects the Upload the Planning Officers Report link
```
[the remark this answers is not in the document]
Comment (pippalou) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:8]: @shilpabasu22 good with me.  Ill change the A/C

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature (hunk 4 above)
[the remark this answers is not in the document]
Comment (pippalou) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:31]: Agreed, I have commented on this above. I think the And statement has just been copied and pasted incorrectly.  It should be as follows: And upload the Planning Officer's report subsection is shown as completed

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js (hunk 2 above)
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:31]: Edited in place here as refactors need to be on seperate stories.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature (hunk 4 above)
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:31]: Done

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature (hunk 5 above)
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:78]: The equivalent wasn't included in AS-9 so I took my cue from there. @shilpabasu22 ?

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  cy.clickChooseFile(fileName);
+  cy.clickUploadFile().click();
+  cy.validateFileUpload(fileName);
+  cy.clickSaveAndContinue();
+});
+
+When('LPA Planning Officer selects the Upload the Planning Officers Report link', () => {
+  cy.goToPage(pageUrl);
+  cy.verifyPage(pageUrl);
+});
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:67]: 1) Presuming this relates to line 65 - Going to a page and checking the links on the main page are two separate tests. Using goToPage() saves ~50 lines of code.

2) It's not requested in the feature file, and wasn't used in AS-9. Happy to introduce it as a new addition.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+const validFileName = 'upload-file-valid.pdf';
+const completedStatusId = 'officers-report';
+const questionTitle = 'Planning Officer’s report';
+
+Given('LPA Planning Officer is within the LPA Questionnaire task list', () => {
+  cy.goToAppealsQuestionnaireTasklistPage();
+});
+
+Given('Upload the Planning Officers report is requested', () => {
+  cy.goToPage(pageUrl);
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:16]: See previous.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+Then LPA Planning Officer is presented with the ability to upload the Planning Officers report
+
+Scenario Outline: AC2 LPA Planning Officer successfully uploads file via upload button
+Given Upload the Planning Officers report is requested
+When a '<valid_file>' is successfully uploaded
+Then progress is made to the Task List
+And Upload the Planning Officers report is shown as completed
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:15]: Done

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+Examples:
+|valid_file|
+|upload-file-valid.tiff|
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:18]: Done

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature (hunk 4 above)
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:31]: Done.

File: lpa-submissions-e2e-tests/cypress/support/appeals-questionnaire-navigation/goToUploadPlanningOfficersReportPage.js [hunk 11]
```diff
@@ -1,8 +0,0 @@
- /// <reference types = "Cypress"/>
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/support/appeals-questionnaire-navigation/goToUploadPlanningOfficersReportPage.js:1]: Agreed, that was an error. This page is deleted now anyway.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js (hunk 1 above)
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:47]: No, this only test the UI process, not actual deletion.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js (hunk 3 above)
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport/appealsPlanningOfficersReportSteps.js:91]: Done.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature (hunk 6 above)
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:8]: Seen.

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature (hunk 4 above)
Comment (pippalou) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:31]: Great

[the remark this answers is not in the document]
Comment (pippalou) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:31]: Thank you

**Review by pippalou — COMMENTED** [review]
Thanks for making the changes, once Shilpa is happy im ok for it to be approved

File: lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature (hunk 5 above)
Comment (shilpabasu22) [reply lpa-submissions-e2e-tests/cypress/integration/appealsPlanningOfficersReport.feature:78]: it wasn't in original acceptance criteria, if you add these @pippalou , we can update the tests

**Review by shilpabasu22 — APPROVED** [review]

**Pull request merged**